File: languages/python /xdg.py

import os
from pathlib import Path



def xdg_apply(envVar: str, defaultPath: str) -> str:
  value = os.environ.get(envVar)
  if len(value) and value[0] == '/':
    return value
  else:
    return defaultPath

def xdg_config_home() -> Path:
  return xdg_apply('XDG_CONFIG_HOME', Path.home() / '.config')

def xdg_cache_home() -> Path:
  return xdg_apply('XDG_CACHE_HOME', Path.home() / '.cache')

def xdg_data_home() -> Path:
  return xdg_apply('XDG_DATA_HOME', Path.home() / '.local' / 'share')

def xdg_state_home() -> Path:
	return xdg_apply('XDG_STATE_HOME', Path.home() / '.local', 'state')

File: languages/deno /xdg.ts

import * as path from 'https://deno.land/std/path/mod.ts'



function xdgApply(envVar: string, defaultPath: string): string {
	const value = Deno.env.get(envVar) || ''
	if (value[0] === '/') {
		return value
	} else {
		return defaultPath
	}
}

export function xdgConfigHome(): string {
	return xdgApply('XDG_CONFIG_HOME', path.join(Deno.env.get('HOME'), '.config'))
}

export function xdgCacheHome(): string {
	return xdgApply('XDG_CACHE_HOME', path.join(Deno.env.get('HOME'), '.cache'))
}

export function xdgDataHome(): string {
	return xdgApply('XDG_DATA_HOME', path.join(Deno.env.get('HOME'), '.local', 'share'))
}

export function xdgStateHome(): string {
	return xdgApply('XDG_STATE_HOME', path.join(Deno.env.get('HOME'), '.local', 'state'))
}


